libamqpprox/amqpprox_session.cpp

```diff
@@ -403,6 +409,46 @@ void Session::establishConnection()
         return;
     }
 
+    auto self(shared_from_this());
+    auto authResponseCb = [this,
+                           self](const AuthResponseData &authResponseData) {
+        if (authResponseData.getAuthResult() ==
+            AuthResponseData::AuthResult::DENY) {
+            LOG_ERROR << "Disconnecting unauthenticated/unauthorized client, "
+                         "reason: "
+                      << authResponseData.getReason();
+            disconnectUnauthClientGracefully();
+            return;
+        }
+        else if (authResponseData.getAuthResult() ==
+                 AuthResponseData::AuthResult::ALLOW) {
+            LOG_TRACE << "Authenticated/Authorized client, reason: "
+                      << authResponseData.getReason();
+            if (!authResponseData.getAuthMechanism().empty() &&
+                !authResponseData.getCredentials().empty()) {
+                d_connector.setAuthMechanismCredentials(
+                    authResponseData.getAuthMechanism(),
+                    authResponseData.getCredentials());
+            }
+        }
+        else {
+            LOG_FATAL << "Not able to authn/authz client. Disconnecting "
+                         "client. Invalid response values from auth gate "
+                         "service, isAllowed: "
+                      << static_cast<int>(authResponseData.getAuthResult())
+                      << ", reason: " << authResponseData.getReason();
+            disconnectUnauthClientGracefully();
+            return;
+        }
+    };
+    const std::pair<std::string_view, std::string_view> credentials =
+        d_connector.getAuthMechanismCredentials();
+
+    d_authIntercept->sendRequest(
+        AuthRequestData(d_sessionState.getVirtualHost(),
+                        credentials.first,
+                        credentials.second),
+        authResponseCb);
     attemptConnection(connectionManager);
```

Comment:
Do we not want this in the callback? We don't really want to even try connecting until after the auth response comes back as allowed?